pkg/metrics/server/server.go

```diff
@@ -249,7 +253,7 @@ func (s *defaultServer) Start(ctx context.Context) error {
 
 func (s *defaultServer) createListener(ctx context.Context, log logr.Logger) (net.Listener, error) {
 	if !s.options.SecureServing {
-		return net.Listen("tcp", s.options.BindAddress)
+		return s.options.ListenConfig.Listen(context.Background(), "tcp", s.options.BindAddress)
```

Comment:
Should we use the context that is passed in?

Reply:
Also, this will always be the listener returned from the `options.ListenConfig`.  Did we not want to fall back on `net.Listen` if that isn't set?

Reply:
@troy0820 To both of your points 

net.Listen definition looks like
```
package net
...
func Listen(network, address string) (Listener, error) {
	var lc ListenConfig
	return lc.Listen(context.Background(), network, address)
}
```

1. I was not trying to change current behavior as you can see net package function Listen just uses Background Context so I assumed the behavior we would want to continue
2. This by design auto falls back I believe but could be wrong.

Reply:
What I mean is that if the option doesn't exist, there is not listener to do this with.  There is no default `net.Listen(...)` because we have to use the `options.ListenConfig` that isn't there if it is not passed through.  Before with the old implementation, we didn't use an option.  If you check that the option exists, you could utilize the pattern you wish.

Reply:
@troy0820 I'm a little confused are you concerned about scope of `var lc ListenConfig` being instantiated inside the built in net.Listen function to have true backwards compatibility? If so I can add an if statement.

Reply:
I was trying to not make s.options.ListenConfig a pointer and without it being a pointer its hard to know if its been set because it is always set. Are you suggesting that we make it a pointer?

Reply:
There are two ways we can solve it being an if statement.

Reply:
I chose option 2 because there are other config the consumer might want to change within ListenConfig